osquery/core/windows/kobjhandle.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#define _WIN32_DCOM
+
+#include "Kobjhandle.h"
+#include "ntapi.h"
+
+#include <Windows.h>
+
+namespace osquery {
+namespace tables {
+
+KObjHandle::KObjHandle() {
+  _h = NULL;
+}
+
+KObjHandle::~KObjHandle() {
+  close();
+}
+
+HANDLE KObjHandle::getAsHandle() {
```

Comment:
This is a neat idea. I think it might make the usage more elegant for sure, @scottlundgren if you get a chance I'd say land something like this, but also I'm fine if we just stick with a `get`er method.